assembler.ts

```diff
@@ -473,6 +474,8 @@ function pseudoPass(behavior: RawBehavior, labelInfo: LabelInfo): Pass {
           }
           behavior.pnames ??= [];
           behavior.pnames[reg.reg - 1] = str(instr, name);
+          behavior.parameters ??= [];
+          behavior.parameters[reg.reg - 1] ??= true;
```

Comment:
I'd prefer to set it to false here, and set things to true at the `case "regRef":` in `resolveLabelsPass`.
As you said, always setting it to true is probably safe. But maybe add a TODO to check `outargs`?

Reply:
Thanks for the pointer, I've implemented this in `resolveLabelsPass`. I'm not 100% about adding another callback for marking the parameter as rw. Could also pass the array, or perhaps reintroduce this method in the `Assembler` scope?